lib/ts/recipe/emailpassword/types.ts

```diff
@@ -246,6 +256,34 @@ export type APIInterface = {
               | {
                     status: "EMAIL_ALREADY_EXISTS_ERROR";
                 }
+              | {
+                    status: "SIGNUP_NOT_ALLOWED";
+                }
+              | GeneralErrorResponse
+          >);
+
+    postSignInAccountLinkingPOST:
+        | undefined
+        | ((input: {
+              formFields: {
+                  id: string;
+                  value: string;
+              }[];
+              session: SessionContainerInterface;
+              options: APIOptions;
+              userContext: any;
+          }) => Promise<
+              | {
+                    status: "OK";
+                    user: User;
+                    session: SessionContainerInterface;
+                }
+              | {
+                    status: "ACCOUNT_LINK_FAILURE";
+                    reason: string;
+                    contactSupport: boolean;
```

Comment:
- contactSupport should not be a boolean.
- We should propagate all the reasons for account link failure to this function? Cause functions like linkAccounts return all of those status types?